### ISSUE TITLE
Feature: Support Pydantic field type List[Vector())

### ISSUE THREAD

**Issue by janaka** [lancedb/lancedb#1838]
### SDK

Python

### Description

I'm trying out LanceDB with Colbert style multi-vectors (actually ColPali). I'm using Pydantic models. I thought a `List[Vector(dim=128)]` would be the best approach but that doesn't seem to be supported by the schema conversion logic for table creation. 

I think a `List[List[float]]` is my workaround - not tried it yet. Some might argue it's not a workaround. Given there's a special type for vector fields I would argue it is a workaround. 

It would be great to have first-class support for multi-vectors.

Trying to create a table from a Pydantic model that has a field of type List[Vector()] isn't supported at present (throws an error). A field directly defined as `Vector()` is supported.

From what i can tell this is because `_py_type_to_arrow_type()` doesn't handle `Vector()`. So when the `list` type handling case recurses child types there's isn't a case to handle `Vector()`. Vector as a top-level type is handled in `_pydantic_to_arrow_type()` which is called first on each field. Then it calls `_py_type_to_arrow_type()` for a subset of fields.

Thanks for considering this.

**Comment by eddyxu** [comment]
@janaka yes, we are thinking to add the support to have this 1:N relationship in each row https://github.com/lancedb/lance/issues/2951

**Comment by janaka** [comment]
Looks like the Vector() nullable option is in the latest release which depends on a pre-release of Lance which isn't published?
